### PR TITLE
Pier co/chore/tdkn 297 disable message output

### PR DESCRIPTION
**Pull request by pierCo**
**What is the problem this Pull Request is trying to solve?**

A new proposal of how to integrate the TDKN-297 feature.
 
**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDKN-297
 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-audit/README.adoc [hunk 1]
```diff
@@ -250,10 +250,6 @@ NOTE: If application is running it needs to be restarted for this change to take
 
 === Disable audit message in audit event
 
-You may use constant `org.talend.logging.audit.impl.AbstractAuditLoggerBase.MDC_OUTPUT_MESSAGE` in the MDC to disable
-message output in case all interesting information is located in MDC.
+It is possible for an `AbstractBackend` to dis  baled log messages if it doesn't need it (e.g. performance consideration).
```
Comment (anicolas) [inline daikon-audit/README.adoc:253]:
```suggestion
It is possible for an `AbstractBackend` to disable log messages if it doesn't need it (e.g. performance consideration).
```

File: daikon-audit/README.adoc [hunk 2]
```diff
@@ -250,10 +250,6 @@ NOTE: If application is running it needs to be restarted for this change to take
...
 
-```
-Context ctx = ContextBuilder.create(MDC_OUTPUT_MESSAGE, "false").build();
-
-```
\ No newline at end of file
+To do that, the implementation of the `AbstractBackend` have to override the method `isOnHoldForFormattedMessages` in order to retuen `false` (default value is `true` - backward compatibility conciderations).
```
Comment (fhuaulme) [inline daikon-audit/README.adoc:255]: Typo on "return"

Comment (fhuaulme) [inline daikon-audit/README.adoc:255]: I would rename method from `isOnHoldForFormattedMessages` to `enableMessageFormat`

**Review by fhuaulme — CHANGES_REQUESTED** [review]

**Review by fhuaulme — APPROVED** [review]
LGTM

**Pull request merged**